File: easy/097_FindAWriter/php/src/findAWriter.php
<?php
//new findWriters($argv[1]);

class findWriters {
    function __construct($file) {
        $fileToParse = fopen($file, 'r');
        while (!feof($fileToParse)) {
            $writer = new findAWriter(fgets($fileToParse));
            $writer->printDecodedMessage();
        }
    }
}

class findAWriter {
    private $inputString = '';
    private $encodedMessage = array();
    private $cipher = array();
    public $decodedMessage = '';

    function __construct($input) {
        $this->inputString = $input;
        $this->parseInput();
        $this->decode();
    }

    function parseInput() {
        $splitInput = explode('|', $this->inputString);
        $this->encodedMessage = str_split($splitInput[0]);
        $this->cipher = explode(' ', $splitInput[1]);
    }
    
    function decode() {
        foreach ($this->cipher as $msg) {
            $this->decodedMessage .= $this->encodedMessage[intval($msg)-1];
        }
    }
    
    function printDecodedMessage() {
        echo $this->decodedMessage . "\n";
    }
}

File: easy/097_FindAWriter/python/FindAWriter.py
import sys

class FindAWriter:
    
    def __init__(self, msg):
        self.cipher = ''
        self.decodedMessage = ''
        self.encodedMessage = []
        
        self.parse(msg)
        self.decode()
        
    def parse(self, msg):
        encoded_message, self.cipher = msg.split('| ')
        self.encodedMessage = list(encoded_message)
        
    def decode(self):
        for num in self.cipher.split(' '):
            self.decodedMessage += self.encodedMessage[int(num)-1]


class FindWriters:
    
    def __init__(self, inputs):
        for line in inputs:
            print(FindAWriter(line).decodedMessage)
        inputs.close()


FindWriters(open(sys.argv[1], 'r'))
